src/noisepy/seis/noise_module.py

```diff
@@ -324,96 +327,114 @@ def stats2inv(stats, prepro_para, locs=None):
     staxml = prepro_para["stationxml"]
     respdir = prepro_para["respdir"]
     input_fmt = prepro_para["input_fmt"]
-
     if staxml:
-        if not respdir:
-            raise ValueError("Abort! staxml is selected but no directory is given to access the files")
-        else:
-            invfilelist = glob.glob(os.path.join(respdir, "*" + stats.station + "*"))
-            if len(invfilelist) > 0:
-                invfile = invfilelist[0]
-                if len(invfilelist) > 1:
-                    print(
-                        (
-                            "Warning! More than one StationXML file was found for station %s."
-                            + "Keeping the first file in list."
-                        )
-                        % stats.station
+        return stats2Inv_staxml(stats, respdir)
+    if input_fmt == "sac":
+        return stats2inv_sac(stats)
+    elif input_fmt == "mseed":
+        return stats2inv_mseed(stats, locs)
+
+
+def stats2Inv_staxml(stats, respdir) -> Inventory:
+    if not respdir:
+        raise ValueError("Abort! staxml is selected but no directory is given to access the files")
+    else:
+        invfilelist = glob.glob(os.path.join(respdir, "*" + stats.station + "*"))
+        if len(invfilelist) > 0:
+            invfile = invfilelist[0]
+            if len(invfilelist) > 1:
+                print(
+                    (
+                        "Warning! More than one StationXML file was found for station %s."
+                        + "Keeping the first file in list."
                     )
-                if os.path.isfile(str(invfile)):
-                    inv = obspy.read_inventory(invfile)
-                    return inv
-            else:
-                raise ValueError("Could not find a StationXML file for station: %s." % stats.station)
+                    % stats.station
+                )
+            if os.path.isfile(str(invfile)):
+                inv = obspy.read_inventory(invfile)
+                return inv
+        else:
+            raise ValueError("Could not find a StationXML file for station: %s." % stats.station)
+
 
+def stats2inv_sac(stats):
     inv = Inventory(networks=[], source="homegrown")
+    net = Network(
+        # This is the network code according to the SEED standard.
+        code=stats.network,
+        stations=[],
+        description="created from SAC and resp files",
+        start_date=stats.starttime,
+    )
 
-    if input_fmt == "sac":
-        net = Network(
-            # This is the network code according to the SEED standard.
-            code=stats.network,
-            stations=[],
-            description="created from SAC and resp files",
-            start_date=stats.starttime,
-        )
+    sta = Station(
```

Comment:
Potentially, although that would lose some of the semantics (e.g. channels are part of a station). It would also duplicate data since all channels belonging to the same station would have the same lat/lon. 

Als note that the above Station and Network types are `obspy` types. I added noisepy specific Channel and Station types to use in the noisepy code so to avoid increasing the dependency on obspy.